crates/contract_history/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+pub const fn current_mainnet() -> &'static [u8] {
+    version_2_2_0()
+}
+
+pub const fn current_testnet() -> &'static [u8] {
+    version_2_2_0()
+}
+
+pub const fn version_2_2_0() -> &'static [u8; 566653] {
+    include_bytes!("../archive/signer-2_2_0.wasm")
+}
+
+#[cfg(test)]
+#[cfg(feature = "external-services-tests")]
+mod tests {
+    use super::*;
+
+    use base64::{Engine as _, prelude::BASE64_STANDARD};
+
+    #[tokio::test]
+    async fn mainnet_contract_should_be_up_to_date() {
+        let on_chain_right_now =
+            fetch_contract_code("https://rpc.mainnet.near.org", "v1.signer").await;
+
+        assert_eq!(current_mainnet(), on_chain_right_now);
```

Comment:
I think comparing the hashes should be sufficient.

Reply:
is there a way to get the hash directly from the rpc? In that case it would make sense :) Else I don't see the advantage

Reply:
just tested, if that hash is the hash of the code, then we can also get the hash only using  https://docs.near.org/api/rpc/contracts#view-account